nscplugin/src/main/scala-3/scala/scalanative/nscplugin/NirGenExpr.scala

```diff
@@ -78,7 +79,35 @@ trait NirGenExpr(using Context) {
       }
     }
 
+    def defnNir(using Context): NirDefinitions = NirDefinitions.get
+
+    lazy val SafeZoneHandle = new Property.Key[Val]
+
+    private def safeZoneHandle(tree: Tree): Val = {
+      if tree.hasAttachment(SafeZoneHandle) then
+        tree.getAttachment(SafeZoneHandle).get
+      else Val.Null
+    }
+
     def genApply(app: Apply): Val = {
+      app match {
+        case Apply(_, List(sz, tree))
+            if app.fun.symbol == defnNir.SafeZoneCompat_withSafeZone && !tree
+              .hasAttachment(SafeZoneHandle) =>
```

Comment:
Is there a case when we would get into `withSafeZone` and tree would have this attachment? 
If no, maybe we can simplify this condition, and add a warning/error when overriding the attachment

Reply:
`withSafeZone(sz, new T(...))` (this `new T(...)` tree has no attachment) is translated to `{ sz.checkOpen(); new T(...)}` (this `new T(...)` tree has attachment). So the condition is needed here.